test/e2e/proxy_test.go

```diff
@@ -53,40 +53,55 @@ func proxyTestSuite(t *testing.T) {
 	require.NoError(t, tc.loadExistingNodes())
 
 	t.Run("Trusted CA ConfigMap validation", tc.testTrustedCAConfigMap)
-	t.Run("Environment variables validation", tc.testEnvVars)
+	// Certificate validation test must run before environment variables removal validation since the latter results in
+	// the deletion of TrustedCAConfigMap, which the former relies on
 	t.Run("Certificate validation", tc.testCerts)
+	t.Run("Environment variables validation", tc.testEnvVars)
 }
 
 // testCerts tests that any additional certificates from the proxy's trusted bundle are imported by each node
 func (tc *testContext) testCerts(t *testing.T) {
-	cm, err := tc.client.K8s.CoreV1().ConfigMaps(wmcoNamespace).Get(context.TODO(),
-		certificates.ProxyCertsConfigMap, meta.GetOptions{})
-	require.NoErrorf(t, err, "error getting trusted CA ConfigMap: %w", err)
+	// TODO: this only tests the user-provided certs, a subset of the required proxy certificates.
+	// Should be addressed with https://issues.redhat.com/browse/WINC-1144
+	cm, err := tc.client.K8s.CoreV1().ConfigMaps(userCABundleNamespace).Get(context.TODO(), userCABundleName, meta.GetOptions{})
+	require.NoErrorf(t, err, "error getting user-provided CA ConfigMap: %w", err)
 
-	// Read all certs from CM data
+	// Read all expected certs from CM data
 	trustedCABundle := cm.Data[certificates.CABundleKey]
 	assert.Greater(t, len(trustedCABundle), 0, "no additional user-provided certs in bundle")
 
-	certs := x509.NewCertPool()
-	require.True(t, certs.AppendCertsFromPEM([]byte(trustedCABundle)), "unable to parse certs from trusted CA ConfigMap data")
-	subjects := certs.Subjects()
-	// Ensure each cert has been imported into every Windows instance's system store
 	for _, node := range gc.allNodes() {
 		t.Run(node.GetName(), func(t *testing.T) {
 			addr, err := controllers.GetAddress(node.Status.Addresses)
 			require.NoError(t, err, "unable to get node address")
 
-			for i, subjectBytes := range subjects {
-				command := fmt.Sprintf("(Get-ChildItem -Path Cert:\\LocalMachine\\Root | "+
-					"Where-Object {$_.Subject -eq '%s'}).Count", string(subjectBytes))
-				out, err := tc.runPowerShellSSHJob(fmt.Sprintf("get-cert-%d", i), command, addr)
+			// Read in one cert at a time and test it exists in the Windows instance's system store
+			scanner := bufio.NewScanner(strings.NewReader(trustedCABundle))
```

Comment:
Have you looked at https://pkg.go.dev/encoding/pem#Decode?
It looks like this functionality might be built in.
```
    for block, rest := pem.Decode(data); block != nil; block, rest = pem.Decode(rest) {
        if  block.Type == "CERTIFICATE"{
```
Bringing this up because this code + splitAtPEMCert() seems to be duplicated from the WICD code. Keeping things simpler here and using stdlib functions might make things easier to maintain.

Not sure if decoding the actual data causes a problem for this. Re-encoding the individual blocks with `pem.Encode` before doing the comparison is always an option :)

Reply:
Good point, I initially was using pem.Decode in the source code but had to move away from that approach to avoid reading an arbitrary number of certs into memory. I don't see any issue using it in the test code (where we are controlling the # of certs)